eisbuk-admin/src/components/atoms/AttendanceSheet/AttendanceSheetSlot.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import TableBody from "@material-ui/core/TableBody";
+import TableRow from "@material-ui/core/TableRow";
+import Table from "@material-ui/core/Table";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import {
+  CustomerWithAttendance,
+  SlotInterface,
+  SlotInterval,
+} from "@/types/temp";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+/** @TODO remove this and import same type when it's merged*/
+export interface Props extends SlotInterface {
+  /**
+   * Record of customers who have booked (or are manually as attended),
+   * keyed by `customerId` and having all of the data for the customer plus
+   * values for `bookedInterval` and `attendedInterval`.
+   */
+  customers: CustomerWithAttendance[];
+}
+
+/** function that calculates start and end time for slot
+ * @param intervals - slot intervals array
+ * @returns overallInterval - { firstStartTime, lastEndTime}
+ */
+const getOverallInterval = (
+  intervals: SlotInterval[]
+): {
+  overallStartTime: string;
+  overallEndTime: string;
+} => {
+  return {
+    overallStartTime: intervals[0].startTime,
+    overallEndTime: intervals[intervals.length - 1].endTime,
+  };
+};
```

Comment:
I would name this `getTimeString` or `getSlotTimespan` and you can return the final string here (instead of start/end time object) so the in-component code is a tad cleaner.